src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -5103,6 +5103,228 @@ class StubGenerator: public StubCodeGenerator {
+   */
+  address generate_base64_encodeBlock() {
+    static const char toBase64[64] = {
+      'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+      'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+      'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+      'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+      '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '+', '/'
+    };
+
+    static const char toBase64URL[64] = {
+      'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+      'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+      'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+      'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+      '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '-', '_'
+    };
+
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", "encodeBlock");
+    address start = __ pc();
+
+    Register src = c_rarg0;
+    Register soff = c_rarg1;
+    Register send = c_rarg2;
+    Register dst = c_rarg3;
+    Register doff = c_rarg4;
+    Register isURL = c_rarg5;
+
+    Register codec = c_rarg6;
+    Register length = c_rarg7; // total length of src data in bytes
+
+    Label ProcessData, Exit, Greater;
+
+    RegSet saved_regs;
+    __ push_reg(saved_regs, sp);
+
+    // length should be multiple of 3
+    __ sub(length, send, soff);
+    __ mv(t0, 3);
+    __ div(length, length, t0);
+    // real src/dst to process data
+    __ add(src, src, soff);
+    __ add(dst, dst, doff);
+
+    // load the codec base address
+    __ la(codec, ExternalAddress((address) toBase64));
+    __ beqz(isURL, ProcessData);
+    __ la(codec, ExternalAddress((address) toBase64URL));
+    __ BIND(ProcessData);
+
+    Label ProcessScalar, ProcessM2, ProcessM1;
+
+    if (!UseRVV) {
+      __ j(ProcessScalar);
+    }
+
+    // vector version
+    {
+      Register tmp1 = x28; // t3
+      Register tmp2 = x29; // t4
+      Register limitM1 = x30; // t5
+      Register limitM2 = x31; // t6
+
+      __ mv(limitM1, MaxVectorSize);
+      __ slli(limitM2, limitM1, 1);
+
+      __ BIND(ProcessM2);
+      __ blt(length, limitM1, ProcessScalar);
+
+      __ blt(length, limitM2, ProcessM1);
+
+      encodeVector(src, dst, codec, limitM2,
+                    v2, v4, v6,         // inputs
+                    v8, v10, v12, v14,  // indexes
+                    v16, v18, v20, v22, // outputs
+                    Assembler::m2);
+
+      __ sub(length, length, limitM2);
+      __ j(ProcessM2);
+
+      __ BIND(ProcessM1);
+      encodeVector(src, dst, codec, limitM1,
+                    v1, v2, v3,         // inputs
+                    v4, v5, v6, v7,     // indexes
+                    v8, v9, v10, v11,   // outputs
+                    Assembler::m1);
+      __ sub(length, length, limitM1);
+    }
+
+    // scalar version
+    __ BIND(ProcessScalar);
```

Comment:
You can move that in the previous block at https://github.com/openjdk/jdk/pull/19973/files#diff-97f199af6d1c8c17b2fa4f50eb1bbc0081858cc59a899f32792a2d31f933ccc4R5260 as it's the only block where it's used.

Reply:
That `ProcessScalar` Label is only ever jumped to if we're in the `UseRVV` block, so please move that `__ BIND(ProcessScalar)` to L5256, inside the `if (UseRVV) { ... }` block.

Reply:
I see, You're right!